src/intrinsics/ecma262/Array.js

```diff
@@ -85,7 +86,12 @@ export default function (realm: Realm): NativeFunctionValue {
         // c. Let intLen be 1.
         intLen = 1;
       } else { // 7. Else,
-        len = len.throwIfNotConcreteNumber();
+        // len = recoverableError(() => len.throwIfNotConcreteNumber(), ()=>new NumberValue(realm, 0));
+        len = recoverableError(
```

Comment:
is this just to get a pull request out?
What's the bigger vision of throwing exceptions vs recoverable errors?
How is it going to be tested?